llvm/lib/Target/AMDGPU/Disassembler/AMDGPUDisassembler.cpp

```diff
@@ -2184,8 +2188,7 @@ AMDGPUDisassembler::decodeKernelDescriptorDirective(
                       KERNEL_CODE_PROPERTY_ENABLE_WAVEFRONT_SIZE32);
     }
 
-    // FIXME: We should be looking at the ELF header ABI version for this.
-    if (AMDGPU::getDefaultAMDHSACodeObjectVersion() >= AMDGPU::AMDHSA_COV5)
+    if (CodeObjectVersion >= AMDGPU::AMDHSA_COV5)
```

Comment:
What is the difference between AMDHSA_COV5 and ELFABIVERSION_AMDGPU_HSA_V5? Can't we use ELFABIVERSION_AMDGPU_HSA_V5 here?

Reply:
`ELFABIVERSION_AMDGPU_HSA_V5` is the value encoded in the ELF header, which is 3 for COV5 and 2 for COV4. `AMDHSA_COV5` is the actual version, so it has value 5. `getELFABIVersion()` maps from actual to encoded version, and `getAMDHSACodeObjectVersion()` maps from encoded to actual version. It's unfortunate that there's a difference :/